contracts/core/Registry.sol

```diff
@@ -252,12 +251,15 @@ contract Registry is IRegistry, Native, AccessControl, Transfer {
     /// ======== Internal Functions ========
     /// ====================================
 
-    /// @notice Generates the anchor for the given identityId and name
+    /// @notice Generates and deploy the anchor for the given identityId and name
     /// @param _identityId Id of the identity
     /// @param _name The name of the identity
-    function _generateAnchor(bytes32 _identityId, string memory _name) internal pure returns (address) {
-        bytes32 attestationHash = keccak256(abi.encodePacked(_identityId, _name));
-        return address(uint160(uint256(attestationHash)));
+    function _generateAnchor(bytes32 _identityId, string memory _name) internal returns (address) {
+        bytes32 salt = keccak256(abi.encodePacked(_identityId, _name));
+
+        bytes memory creationCode = abi.encodePacked(type(Anchor).creationCode, abi.encode(_identityId));
+
+        return CREATE3.deploy(salt, creationCode, 0);
```

Comment:
need to clarify what the salt should be

Reply:
> need to clarify what the salt should be

i think identityId and name are fine. it's like we had before